java/client/src/main/java/glide/api/GlideClusterClient.java

```diff
@@ -106,21 +106,46 @@ public CompletableFuture<ClusterValue<Object>> customCommand(@NonNull String[] a
                 CustomCommand, args, response -> ClusterValue.of(handleObjectOrNullResponse(response)));
     }
 
+    @Override
+    public CompletableFuture<ClusterValue<Object>> customCommand(@NonNull GlideString[] args) {
+        // TODO if a command returns a map as a single value, ClusterValue misleads user
+        return commandManager.submitNewCommand(
+                CustomCommand,
+                args,
+                response -> ClusterValue.of(handleBinaryObjectOrNullResponse(response)));
+    }
+
     @Override
     public CompletableFuture<ClusterValue<Object>> customCommand(
             @NonNull String[] args, @NonNull Route route) {
         return commandManager.submitNewCommand(
-                CustomCommand, args, route, response -> handleCustomCommandResponse(route, response));
+                CustomCommand,
+                args,
+                route,
+                response -> handleCustomCommandResponse(route, response, false));
+    }
+
+    @Override
+    public CompletableFuture<ClusterValue<Object>> customCommand(
+            @NonNull GlideString[] args, @NonNull Route route) {
+        return commandManager.submitNewCommand(
+                CustomCommand, args, route, response -> handleCustomCommandResponse(route, response, true));
     }
 
-    protected ClusterValue<Object> handleCustomCommandResponse(Route route, Response response) {
+    protected ClusterValue<Object> handleCustomCommandResponse(
```

Comment:
Convention says we usually don't include isBinaryData in the function, but use overloading.  E.g. `handleBinaryCustomCommandResponse`.